WalletWasabi/WabiSabi/Client/CredentialDependencies/DependencyGraph.cs

```diff
@@ -0,0 +1,443 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Client.CredentialDependencies
+{
+	public record DependencyGraph
+	{
+		public const int K = ProtocolConstants.CredentialNumber;
+
+		public static ImmutableArray<CredentialType> CredentialTypes { get; } = ImmutableArray.Create<CredentialType>(CredentialType.Amount, CredentialType.Vsize);
+
+		public ImmutableList<RequestNode> Vertices { get; private set; } = ImmutableList<RequestNode>.Empty;
+
+		public ImmutableList<RequestNode> Inputs { get; private set; } = ImmutableList<RequestNode>.Empty;
+
+		public ImmutableList<RequestNode> Outputs { get; private set; } = ImmutableList<RequestNode>.Empty;
+
+		public ImmutableList<RequestNode> Reissuances { get; private set; } = ImmutableList<RequestNode>.Empty; // TODO sort
+
+		// Internal properties used to keep track of effective values and edges
+		public ImmutableSortedDictionary<CredentialType, CredentialEdgeSet> edgeSets { get; init; } = ImmutableSortedDictionary<CredentialType, CredentialEdgeSet>.Empty
+			.Add(CredentialType.Amount, new() { CredentialType = CredentialType.Amount })
+			.Add(CredentialType.Vsize, new() { CredentialType = CredentialType.Vsize });
```

Comment:
Credential types are not hardcoded anywhere else what makes me thing the graph can work okay with any number of credential types (I can be very wrong). Is this okay?